gateway-ha/src/main/java/io/trino/gateway/proxyserver/ProxyRequestHandler.java

```diff
@@ -276,12 +277,12 @@ private ProxyResponse recordBackendForQueryId(Request request, ProxyResponse res
         return response;
     }
 
-    public static QueryHistoryManager.QueryDetail getQueryDetailsFromRequest(Request request)
+    public static QueryHistoryManager.QueryDetail getQueryDetailsFromRequest(Request request, String username)
```

Comment:
Can we pass `Optional<String> username`, and then either
```
username.ifPresent(queryDetail::setUser)
```
which changes the current semantics to use `null` instead of `""`, or 
```
queryDetail.setUser(username.orElse(""))
```